### PR TITLE
use quack for rms norm and sft loss

### PR DESCRIPTION
**Pull request by faresobeid**
> [!NOTE]
> **Medium Risk**
> Introduces new CUDA-kernel-backed code paths for SFT loss and RMSNorm that can change numerical behavior and fail at runtime if GPU capability/dependencies don’t match (plus an explicit Gemma incompatibility). Dependency and attention-kernel upgrades add some integration risk but changes are gated behind config/availability checks.
> 
> **Overview**
> Adds a new SFT `loss_impl = "quack_fused"` option that computes fused LM-head + cross-entropy via `quack-kernels` (avoiding full logits materialization), and wires it through `setup_model()`/SFT training so fused losses are handled inside the LM head.
> 
> Extends the LM-head injection logic to support multiple fused implementations (`liger` vs `quack`), centralizes the fused CE ignore index, and explicitly blocks `quack_fused` for Gemma logit softcapping.
> 
> Introduces optional quack-accelerated `RMSNorm` on CUDA (Hopper+) with a guard to force contiguous gradients, and updates packaging by adding a `quack` extra (pinning `quack-kernels>=0.3.3`) and switching `flash-attn-cute` to `flash-attn-4`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 03d5d64a551addda06c8a1e2dad3ae086de221f2. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/prime_rl/trainer/sft/train.py [hunk 1]
```diff
@@ -119,9 +119,8 @@ def train(config: SFTConfig):
     # Initialize the model and tokenizer
     logger.info(f"Initializing model ({config.model})")
     loading_from_ckpt_later = config.ckpt and checkpoint_step is not None
-    model = setup_model(
-        config.model, parallel_dims, loading_from_ckpt_later, fused_cross_entropy=config.loss_impl == "liger_fused"
-    )
+    fused_cross_entropy: bool | str = {"liger_fused": "liger", "quack_fused": "quack"}.get(config.loss_impl, False)
+    model = setup_model(config.model, parallel_dims, loading_from_ckpt_later, fused_cross_entropy=fused_cross_entropy)
```
Comment (samsja) [inline src/prime_rl/trainer/sft/train.py:123]:
```suggestion
    fused_cross_entropy = {"liger_fused": "liger", "quack_fused": "quack"}.get(config.loss_impl, False)
    model = setup_model(config.model, parallel_dims, loading_from_ckpt_later, fused_cross_entropy=fused_cross_entropy)
```

**Review by samsja — COMMENTED** [review]
looks good, but shouldn't we add quack as dependency also then ?

File: pyproject.toml [hunk 2]
```diff
@@ -80,6 +80,9 @@ disagg = [
 ]
 
 [dependency-groups]
+quack = [
+    "quack-kernels>=0.3.3",
+]
```
Comment (samsja) [inline pyproject.toml:83]: lets no add here but in the extra

Comment (samsja) [reply pyproject.toml:83]: like flash-attn-cute

**Review by samsja — APPROVED** [review]
